pkg/eventmonitor/config/config.go

```diff
@@ -34,21 +33,17 @@ type Config struct {
 func NewConfig() *Config {
 	return &Config{
 		// event server
-		SocketPath:       pkgconfigsetup.SystemProbe().GetString(join(evNS, "socket")),
-		EventServerBurst: pkgconfigsetup.SystemProbe().GetInt(join(evNS, "event_server.burst")),
+		SocketPath:       pkgconfigsetup.SystemProbe().GetString(sysconfig.FullKeyPath(evNS, "socket")),
+		EventServerBurst: pkgconfigsetup.SystemProbe().GetInt(sysconfig.FullKeyPath(evNS, "event_server.burst")),
 
 		// consumers
 		ProcessConsumerEnabled: getBool("process.enabled"),
 	}
 }
 
-func join(pieces ...string) string {
-	return strings.Join(pieces, ".")
-}
-
 func getAllKeys(key string) (string, string) {
-	deprecatedKey := strings.Join([]string{rsNS, key}, ".")
-	newKey := strings.Join([]string{evNS, key}, ".")
+	deprecatedKey := sysconfig.FullKeyPath(rsNS, key, ".")
```

Comment:
are you sure about this ? I'm surprised we are keeping the "." here

Reply:
No, my main issue is that
```
strings.Join([]string{rsNS, key}, ".") = `{rsNS}.{key}`
```
while
```
sysconfig.FullKeyPath(rsNS, key, ".") =  `{rsNS}.{key}..`
```

so I think it should just be `sysconfig.FullKeyPath(rsNS, key)` without the dot as the last argument

Reply:
ahh lol, yeah my bad